### PR TITLE
Refactor Readwrite-splitting loadbalance.

### PR DESCRIPTION
**Pull request by zjcnb**

### CODE REVIEW

File: shardingsphere-features/shardingsphere-readwrite-splitting/shardingsphere-readwrite-splitting-core/src/main/java/org/apache/shardingsphere/readwritesplitting/algorithm/loadbalance/AbstractReplicaLoadBalance.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ * limitations under the License.
+ */
+
+package org.apache.shardingsphere.readwritesplitting.algorithm.loadbalance;
+
+import org.apache.shardingsphere.readwritesplitting.spi.ReplicaLoadBalanceAlgorithm;
+
+import java.util.List;
+
+public abstract class AbstractReplicaLoadBalance implements ReplicaLoadBalanceAlgorithm {
```
Comment (terrymanu) [inline shardingsphere-features/shardingsphere-readwrite-splitting/shardingsphere-readwrite-splitting-core/src/main/java/org/apache/shardingsphere/readwritesplitting/algorithm/loadbalance/AbstractReplicaLoadBalance.java:24]: The class name AbstractReplicaLoadBalance is not good enough, `Balance` is verb.
How about rename it to AbstractReplicaLoadBalanceAlgorithm?
If move the logic into `ReadwriteSplittingDataSourceRouter`, we can remove this class.

File: shardingsphere-features/shardingsphere-readwrite-splitting/shardingsphere-readwrite-splitting-core/src/main/java/org/apache/shardingsphere/readwritesplitting/algorithm/loadbalance/WeightReplicaLoadBalanceAlgorithm.java [hunk 2]
```diff
@@ -121,4 +112,9 @@ private double getWeightValue(final String readDataSourceName) {
         }
         return weight;
```
Comment (terrymanu) [inline shardingsphere-features/shardingsphere-readwrite-splitting/shardingsphere-readwrite-splitting-core/src/main/java/org/apache/shardingsphere/readwritesplitting/algorithm/loadbalance/WeightReplicaLoadBalanceAlgorithm.java:113]: rename as `result`

File: shardingsphere-features/shardingsphere-readwrite-splitting/shardingsphere-readwrite-splitting-core/src/main/java/org/apache/shardingsphere/readwritesplitting/algorithm/loadbalance/AbstractReplicaLoadBalance.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    
+    @Override
+    public String getDataSource(final String name, final String writeDataSourceName, final List<String> readDataSourceNames) {
+        if (1 == readDataSourceNames.size()) {
+            return readDataSourceNames.get(0);
+        }
```
Comment (terrymanu) [inline shardingsphere-features/shardingsphere-readwrite-splitting/shardingsphere-readwrite-splitting-core/src/main/java/org/apache/shardingsphere/readwritesplitting/algorithm/loadbalance/AbstractReplicaLoadBalance.java:30]: How about move this logic into `ReadwriteSplittingDataSourceRouter.route()`?
We should avoid to use abstract class if possible.

**Review by terrymanu — CHANGES_REQUESTED** [review]

File: shardingsphere-features/shardingsphere-readwrite-splitting/shardingsphere-readwrite-splitting-core/src/main/java/org/apache/shardingsphere/readwritesplitting/algorithm/loadbalance/AbstractReplicaLoadBalance.java (hunk 1 above)
Comment (zjcnb) [reply shardingsphere-features/shardingsphere-readwrite-splitting/shardingsphere-readwrite-splitting-core/src/main/java/org/apache/shardingsphere/readwritesplitting/algorithm/loadbalance/AbstractReplicaLoadBalance.java:24]: Yes, that's also ok. i will move it.

File: shardingsphere-features/shardingsphere-readwrite-splitting/shardingsphere-readwrite-splitting-core/src/main/java/org/apache/shardingsphere/readwritesplitting/algorithm/loadbalance/WeightReplicaLoadBalanceAlgorithm.java (hunk 2 above)
Comment (zjcnb) [reply shardingsphere-features/shardingsphere-readwrite-splitting/shardingsphere-readwrite-splitting-core/src/main/java/org/apache/shardingsphere/readwritesplitting/algorithm/loadbalance/WeightReplicaLoadBalanceAlgorithm.java]: Done.

**Review by taojintianxia — APPROVED** [review]

**Review by terrymanu — APPROVED** [review]

**Pull request merged**